### PR TITLE
update image for individual song link

### PR DESCRIPTION
**Pull request by TahirHameed74**
update image for individual song link

### CODE REVIEW

**Comment by TahirHameed74** [comment]
Screenshot 2024-06-19 at 6 02 36 PM

Example output of the OG image.

File: components/Profile/Rank.tsx [hunk 1]
```diff
@@ -1,30 +1,18 @@
...
-      if (!profile?.verifications) return;
-      const leaderBoardRank: any = await getProfileRank(profile?.verifications);
-      setRank(leaderBoardRank);
-    };
-    if (!profile) return;
-
-    init();
-  }, [profile]);
+  if (rank === null) {
+    return null;
```
Comment (SweetmanTech) [inline components/Profile/Rank.tsx:7]: please do not return `empty` components.

instead, handle this login in the parent to `conditionally render` the `<Rank />` component

File: app/cast/[username]/[hash]/page.tsx [hunk 2]
```diff
@@ -21,9 +23,54 @@ export const metadata: Metadata = {
...
+      icons: [`${VERCEL_URL}/images/logo2.png`],
+      other: {
+        ...frameMetadata,
+      },
+    };
+  } catch (error) {
+    console.error('Failed to generate metadata:', error);
+    return metadata;
+  }
+}
```
[the remark this answers is not in the document]
Comment (TahirHameed74) [reply app/cast/[username]/[hash]/page.tsx]: This is metadata function which needs to be implemented and exported inside the page. Can you clarify your expectations here?

File: components/Profile/Rank.tsx (hunk 1 above)
Comment (TahirHameed74) [reply components/Profile/Rank.tsx]: This has been fixed.

File: app/cast/[username]/[hash]/page.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (TahirHameed74) [reply app/cast/[username]/[hash]/page.tsx]: I have moved most of the logic outside the component and moved it in utils folder.

File: app/api/og-image/cast/[username]/[hash]/[[...rank]]/route.tsx
Comment (SweetmanTech) [inline app/api/og-image/cast/[username]/[hash]/[[...rank]]/route.tsx:1]: route should not include `rank` param

File: app/api/og-image/cast/[username]/[hash]/[[...rank]]/route.tsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+  const points = formatPoints(cast?.points);
+
+  const soraSemiBold = fetch(new URL('/public/Sora-SemiBold.ttf', import.meta.url)).then((res) =>
+    res.arrayBuffer(),
+  );
+
+  const soraNormal = fetch(new URL('/public/Sora-Regular.ttf', import.meta.url)).then((res) =>
+    res.arrayBuffer(),
+  );
```
Comment (SweetmanTech) [inline app/api/og-image/cast/[username]/[hash]/[[...rank]]/route.tsx:38]: move all this logic into standalone libs
- rule: each function should do exactly one thing
- actual: this function is doing `profile lookup`, `metadata lookup`, `channel lookup`, etc all defined in the same file
- solution: move this logic into a single lib, called by this API endpoint

File: app/api/og-image/cast/[username]/[hash]/[[...rank]]/route.tsx [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+        },
+        {
+          name: 'Sora',
+          data: await soraSemiBold,
+          weight: 600,
+        },
+      ],
+    },
+  );
+}
```
Comment (SweetmanTech) [inline app/api/og-image/cast/[username]/[hash]/[[...rank]]/route.tsx:198]: file too long
- rule: files should be a `less than 100 lines of code`
- actual: @TahirHameed74 file has ~200 lines of code
- suggestion: break this down into smaller libs / reusable components

File: app/cast/[username]/[hash]/page.tsx [hunk 5]
```diff
@@ -21,9 +24,53 @@ export const metadata: Metadata = {
   },
 };
```
Comment (SweetmanTech) [inline app/cast/[username]/[hash]/page.tsx:25]: if you remove the `export` this variable becomes unused right?

Comment (TahirHameed74) [reply app/cast/[username]/[hash]/page.tsx:25]: no i am returning default metadata if cast is not found

 catch (error) {
    console.error('Failed to generate metadata:', error);
    return metadata;
  }

Comment (TahirHameed74) [reply app/cast/[username]/[hash]/page.tsx:25]: so that the og image is not empty.

**Review by SweetmanTech — APPROVED** [review]

**Review by SweetmanTech — APPROVED** [review]

**Pull request merged**